### DISCUSSION TITLE
How do i use ejabberd to notify my application server for all, offline messages?

### DISCUSSION THREAD

**Discussion by joedomv** [processone/ejabberd#4121] [Questions] [answered]
Hi Team,
We are using ejabberd for chat application to send messages to our react native apps. We are using the bosh connection to send messages. We want to notify the users if they are offline. It seems that mod_push is the right solution for this. but dont see any configuration that allows to push to app servers. we feel we am missing something here. Please help!
Thanks

**Comment by licaon-kter** [comment]
Umm don't use bosh, use normal TCP

Read this: https://github.com/processone/ejabberd/issues/4120#issuecomment-1850878221

**Reply by joedomv** [reply]
@licaon-kter thanks for the input. yeah but wanted to understand , who gets notified here. since the client is already offline, it wont be notified, i would ideally like to push it to a server that can then notify the app via APNS or FCM. sorry if i missing something here.

**Reply by licaon-kter** [reply, marked as answer]
A contact client -> contact server -> your server (xmpp) -> your push component -> google (fcm) or apple (apns) -> your device (wakes up your client)
